File: react/src/lib/domain/entities/DataObjectType.ts
import BaseEnum from './BaseEnum';

export default class DataObjectType extends BaseEnum {
    public static COBOL_COPYBOOK = new DataObjectType('1', 'dataObject:dataObjectTypeEnum:cobolCopybook');
    public static JAVA_CLASS = new DataObjectType('2', 'dataObject:dataObjectTypeEnum:javaClass');
    public static DDL = new DataObjectType('3', 'dataObject:dataObjectTypeEnum:ddl');
    public static XML = new DataObjectType('4', 'dataObject:dataObjectTypeEnum:xml');
    public static MAINFRAME_ASSEMBLER = new DataObjectType('5', 'dataObject:dataObjectTypeEnum:mainframeAssembler');

    public constructor(id?: string, name?: string) {
        super(id, name);
    }
}

File: backend/src/main/java/com/equisoft/dca/backend/dataobject/model/DataObjectType.java
package com.equisoft.dca.backend.dataobject.model;

import com.equisoft.dca.backend.common.model.BaseEnum;

import lombok.Getter;

@Getter
public enum DataObjectType implements BaseEnum<Integer> {

	COBOL_COPYBOOK(1, "COBOL Copybook"),
	JAVA_CLASS(2, "Java class"),
	DDL(3, "DDL"),
	XML(4, "XML"),
	MAINFRAME_ASSEMBLER(5, "Mainframe assembler");

	private final int id;

	private final String name;

	DataObjectType(int id, String name) {
		this.id = id;
		this.name = name;
	}

	@Override
	public Integer getId() {
		return id;
	}
}
